pkg/testcase/cluster.go

```diff
@@ -11,39 +11,105 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+var arch string
+
 func TestBuildCluster(g GinkgoTInterface) {
+	var err error
+
 	cluster := factory.GetCluster(g)
+	arch = cluster.ArchType
 	Expect(cluster.Status).To(Equal("cluster created"))
+	Expect(shared.KubeConfigFile).ShouldNot(BeEmpty())
+	Expect(cluster.ServerIPs).ShouldNot(BeEmpty())
 
-	if strings.Contains(cluster.ClusterType, "etcd") {
-		fmt.Println("Backend:", cluster.ClusterType)
-	} else {
-		fmt.Println("Backend:", cluster.ExternalDb)
-	}
-
-	if cluster.ExternalDb != "" && cluster.ClusterType == "" {
-		for i := 0; i > len(cluster.ServerIPs); i++ {
-			cmd := "grep \"datastore-endpoint\" /etc/systemd/system/k3s.service"
-			res, err := shared.RunCommandOnNode(cmd, cluster.ServerIPs[0])
-			Expect(err).NotTo(HaveOccurred())
-			Expect(res).Should(ContainSubstring(cluster.RenderedTemplate))
+	if cluster.ProductType == "k3s" {
+		if strings.Contains(cluster.ClusterType, "etcd") {
+			fmt.Println("Backend:", cluster.ClusterType)
+		} else {
+			fmt.Println("Backend:", cluster.ExternalDb)
+		}
+	
+		if cluster.ExternalDb != "" && cluster.ClusterType == "" {
+			for i := 0; i > len(cluster.ServerIPs); i++ {
+				cmd := "grep \"datastore-endpoint\" /etc/systemd/system/k3s.service"
+				res, err := shared.RunCommandOnNode(cmd, cluster.ServerIPs[0])
+				Expect(err).NotTo(HaveOccurred())
+				Expect(res).Should(ContainSubstring(cluster.RenderedTemplate))
+			}
 		}
 	}
 
-	err := shared.PrintFileContents(shared.KubeConfigFile)
+	fmt.Println("\nKubeconfig file:\n")
+	err = shared.PrintFileContents(shared.KubeConfigFile)
 	if err != nil {
 		return
 	}
 
-	Expect(shared.KubeConfigFile).ShouldNot(BeEmpty())
-	Expect(cluster.ServerIPs).ShouldNot(BeEmpty())
+	fmt.Println("Base64 Encoded Kubeconfig file:\n")
+	err = shared.PrintBase64Encoded(shared.KubeConfigFile)
+	if err != nil {
+		return
+	}
 
 	fmt.Println("Server Node IPS:", cluster.ServerIPs)
-	fmt.Println("Agent Node IPS:", cluster.AgentIPs)
 
-	if cluster.NumAgents > 0 {
-		Expect(cluster.AgentIPs).ShouldNot(BeEmpty())
+	checkAndPrintAgentNodeIPs(cluster.NumAgents, cluster.AgentIPs, false)
+
```

Comment:
what u think of always print  all ? like so 
```
Server Node IPS: [3.140.188.209 3.144.29.56]
Agent Node IPS: [3.137.163.226]
Windows Agent Node IPS: []
```

in this way we dont need to add this if and also dont need to add a function to it and repeat yourself calling the funciont 2 times